lib/gen_stage.ex

```diff
@@ -2313,16 +2339,30 @@ defmodule GenStage do
     stage
   end
 
-  defp buffer_events(events, %{buffer: buffer, buffer_keep: keep} = stage) do
+  defp buffer_events(
+         events,
+         %{
+           mod: mod,
+           buffer: buffer,
+           buffer_keep: keep,
+           mod_handle_buffered: mod_handle_buffered,
+           state: state
+         } = stage
+       ) do
     {buffer, excess, perms} = Buffer.store_temporary(buffer, events, keep)
 
-    case excess do
-      0 ->
-        :ok
+    if mod_handle_buffered do
```

Comment:
hmm, is there a reason you didn't choose to use `defoverridable` here, as genserver does (https://github.com/elixir-lang/elixir/blob/v1.10.3/lib/elixir/lib/gen_server.ex#L807-L823)?

seems that this would be a bunch nicer.